pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -822,3 +834,69 @@ impl<'a> DeltaValueIter<'a> {
         }
     }
 }
+///
+/// Iterator over all keys stored in a delta layer
+///
+/// FIXME: This creates a Vector to hold the offsets of all key-offset pairs.
+/// That takes up quite a lot of memory. Should do this in a more streaming
+/// fashion.
```

Comment:
You could construct a Vec with just one entry per key, instead of one entry for each key+lsn. That would take less memory and be more convenient for the caller, too.

Reply:
Thank you for suggestion.
Fixed